### PR TITLE
DAOS-623 ci: Remove call to checkpatch.pl

### PR DESCRIPTION
**Pull request by jolivier23**
Replacing checkpatch with clang-format

### CODE REVIEW

File: jenkins_github_checkwarn.sh [hunk 1]
```diff
@@ -36,9 +35,9 @@ def_checkpatch_paths="${check_make}"
 if [ "${FULL_REVIEW}" -ne 0 ]; then
   # Skip pylint checking if performed by GitHub.
   if [ -e .github/workflows/pylint.yml ]; then
-    def_checkpatch_paths+=":${check_style}:${check_shell}"
+    def_checkpatch_paths+=":${check_shell}"
```
Comment (jolivier23) [inline jenkins_github_checkwarn.sh:38]: Perhaps I should do something similar with clang-format that we do with pylint

Comment (ashleypittman) [reply jenkins_github_checkwarn.sh:38]: What do you mean?

Comment (jolivier23) [reply jenkins_github_checkwarn.sh:38]: Like replicate line 37 and add check_style where clang-format isn't used.

Comment (ashleypittman) [reply jenkins_github_checkwarn.sh:38]: Oh, I see what you mean.  Yes, that would be safest with historic releases but given we have clang-format on master and 2.4 how many PRs will still use checkpatch?

**Comment by jolivier23** [comment]
I have no idea how to fix the test.  In theory, it hasn't changed in this repo.

**Comment by jolivier23** [comment]
> I have no idea how to fix the test. In theory, it hasn't changed in this repo.

Looks like prior patch failed in the same way so likely been broken for a while.

**Review by ashleypittman — APPROVED** [review]

**Pull request merged**